packages/base/src/widget.ts

```diff
@@ -991,67 +1000,61 @@ export class DOMWidgetView extends WidgetView {
     this.updateTooltip();
   }
 
-  setLayout(layout: LayoutModel, oldLayout?: LayoutModel): void {
+  async setLayout(layout: LayoutModel, oldLayout?: LayoutModel): Promise<void> {
     if (layout) {
-      this.layoutPromise = this.layoutPromise.then((oldLayoutView) => {
-        if (oldLayoutView) {
-          oldLayoutView.unlayout();
-          this.stopListening(oldLayoutView.model);
-          oldLayoutView.remove();
-        }
+      const oldLayoutView = await this.layoutPromise;
+      if (oldLayoutView) {
+        oldLayoutView.unlayout();
+        this.stopListening(oldLayoutView.model);
+        oldLayoutView.remove();
+      }
 
-        return this.create_child_view(layout)
-          .then((view) => {
-            // Trigger the displayed event of the child view.
-            return this.displayed.then(() => {
-              view.trigger('displayed');
-              this.listenTo(view.model, 'change', () => {
-                // Post (asynchronous) so layout changes can take
-                // effect first.
-                MessageLoop.postMessage(
-                  this.luminoWidget,
-                  Widget.ResizeMessage.UnknownSize
-                );
-              });
-              MessageLoop.postMessage(
-                this.luminoWidget,
-                Widget.ResizeMessage.UnknownSize
-              );
-              this.trigger('layout-changed');
-              return view;
-            });
-          })
-          .catch(
-            utils.reject('Could not add LayoutView to DOMWidgetView', true)
+      try {
+        const view = await this.create_child_view(layout);
+        // Trigger the displayed event of the child view.
+        await this.displayed;
+        view.trigger('displayed');
+        this.listenTo(view.model, 'change', () => {
+          // Post (asynchronous) so layout changes can take
+          // effect first.
+          MessageLoop.postMessage(
+            this.luminoWidget,
+            Widget.ResizeMessage.UnknownSize
           );
-      });
+        });
+        MessageLoop.postMessage(
+          this.luminoWidget,
+          Widget.ResizeMessage.UnknownSize
+        );
+        this.trigger('layout-changed');
+        this.layoutPromise = Promise.resolve(view);
+      } catch {
+        utils.reject('Could not add LayoutView to DOMWidgetView', true);
+      }
     }
   }
 
-  setStyle(style: StyleModel, oldStyle?: StyleModel): void {
+  async setStyle(style: StyleModel, oldStyle?: StyleModel): Promise<void> {
```

Comment:
Same as earlier, shoudl we be chaning these to be async?
If we are changing them, shoudln't all of the calling code be made `async` as well?

Reply:
Updated this in a similar way to above - The established method of setStyle remains sync. The inner promise is converted to async. This way prevents any confusion if `setStyle` should now be awaited or not.

Also like above, the inverse of the `if (style)` block is used to remove a layer of indenting.